src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java

```diff
@@ -500,11 +464,53 @@ private String queryEmail() {
     return paramMap;
   }
 
+  private Account getActiveAccount() {
+    Preconditions.checkState(isLoggedIn());
+    Preconditions.checkState(!accountRoster.isEmpty());
+
+    return accountRoster.getActiveAccount();
+  }
+
+  /**
+   * Sets an account with the given {@code email} as an active account. Does nothing if a matching
+   * account does not exist. Calls {@link UiFacade#notifyStatusIndicator} at the end.
+   *
+   * @param email cannot be {@code null}.
+   */
+  public void setActiveAccount(String email) {
+    Preconditions.checkNotNull(email);
+
+    if (accountRoster.setActiveAccount(email)) {
+      persistCredentials();
+      notifyLoginStatusChange(true /* not a logout */);
+      uiFacade.notifyStatusIndicator();
+    }
+  }
+
+  /**
+   * Returns a list of currently logged-in accounts. Intended for UI to call for the purpose of
+   * updating login widgets, e.g., inside it inside {@link UiFacade#notifyStatusIndicator}.
+   *
+   * @return a list of logged-in accounts. Never {@code null}. An empty list if there is no
+   *     logged-in account. The first {@link Account} is an active account.
+   */
+  public List<AccountInfo> listAccounts() {
+    return accountRoster.listAccounts();
+  }
+
   private void persistCredentials() {
-    Preconditions.checkState(isLoggedIn);
+    Preconditions.checkState(isLoggedIn());
 
-    OAuthData creds = new OAuthData(
-        accessToken, refreshToken, email, oAuthScopes, accessTokenExpiryTime);
-    authDataStore.saveOAuthData(creds);
+    Account account = getActiveAccount();
+    OAuthData oAuthData = new OAuthData(account.getAccessToken(), account.getRefreshToken(),
+        account.getEmail(), oAuthScopes, account.getAccessTokenExpiryTime());
+    authDataStore.saveOAuthData(oAuthData);
```

Comment:
This is to be done in #23. I see this can be done incrementally, as at least saving/loading one account is working.